types/weixin-app/index.d.ts

```diff
@@ -3460,7 +3488,7 @@ interface Component<T> {
      */
     setData(
         data: {
-            [key in keyof T]?:
+            [key in keyof D]?:
```

Comment:
i want to make sure people separate data from props so that this won't update properties.

but in reality i think you can do it.